### DISCUSSION TITLE
Goals - Use cases, the why

### DISCUSSION THREAD

**Discussion by Teprifer** [actualbudget/releases#117] [Feature ideas]
Couldn't see a thread on this feature so I thought I'd start one to write the Why of the goals I use.

Essentially they're to make budgeting lazy, just go down through priority order and click the button to assign a pre-calculated amount and finish up will allocating the remainder to discretionary categories.

The order of the below has no meaning and is assuming monthly budgeting.

1) (Budget X per number of specific Day of week in month)
I go food shopping once a week(generally) or otherwise think of shopping as having a weekly limit.
I define 'shopping day' as Sunday as that's when I do the main shop, so my budget for the month needs enough it in to cover the number of main shops(Sundays) in the month.

2) (add constant amount)
Eating out, I like to budget a fixed amount per month, not too worried about how high it gets if any roll-over accumulates.
Maintenance! Teeth! You never know how much you need, only that you'll probably never get enough in the category and will have to make _decisions_ in the future, but every little bit helps!

3) (constant amount, until cap)
Clothing, I like to budget a set amount per month, but I don't want this category to end up with too much in it, so the 'set amount' need to be disabled while at/above the cap amount (and would only assign up to the cap if less than the set amount).
For clothing specifically this reflects the fact I'll spend from the category unevenly as time goes on, but the amount that goes in to it should be constant unless the category balance gets too high. Essentially averaging the expense.
Same idea for stuff like health & well being, home and kitchen, other 'wish farm' stuff like new mattress, new TV, etc. where I don't have a time I want the total by, just care about how much I'm putting towards it each month.

4) (full refill to cap)
I have monthly bills where the actual amount fluctuates, think power, water, etc.
I know the bill will never be over X amount, and at the start of each month I want to top up the category to X again.
This can also be looked at the other way, which is to top it up by how much I spent last month. (This requires setting it to the right amount initially for new starters.)

5) (An amount that equals the sum of all scheduled transactions for that category due that month)
I have a mortgage split in to different portions and pay it weekly, this means there's 3 scheduled transactions occurring each week.
Because they're weekly the amount I pay each month differs, there will either be 4 sets of transactions, or 5.
I'd like to one-click set the category to the exact amount due that month based on the sum of all the transactions so the right amount is there to be subtracted from as the month goes on. Ending the month at $0.00 in the category.

This also works for bills/subscriptions that have payment periods not equal to 1 month - I once had a mobile bill that was due every 28 days. Very annoying.

6) (X by due date, same amount per month with optional renewing period length)
I have some yearly bills and I know when they're due so I have a category for each and want to put in the same amount each month (because my income is regular) in to that category such that when the bill is due, the category is fully funded.
Also forget bills, Christmas gift money!
The renewing period length is to reflect that once the first due date is reached, the payment will need to be made again next year so I need to continue budgeting the same amount each month.
Where there's a 'due date' the calculation should be monthly, so if I under categorise once month, then I make up for it in the future months.
Likewise if I over fund one month (such as when it's a 3 fortnight payday month) the amount of over payment is effectively divided by the number of remaining months so that I pay less each of the following months.

Interestingly something like subscriptions can be covered by different months depending on how some people have them set up.
Some like to bundle them all together in to one category so something like 'sum of upcoming transactions in month for the category' may work better that a fixed amount per month.

There's some great comments on this Reddit thread: [https://www.reddit.com/r/actualbudget/comments/qpp8nr/goals_your_use_cases/](https://www.reddit.com/r/actualbudget/comments/qpp8nr/goals_your_use_cases/)

**Comment by Deguello** [comment]
It would be nice to be able to apply all goals at once with a single action, rather than going through each one individually. This would be a pretty straightforward way to set up a monthly template without much coding beyond what's required for the goals, anyway.

**Comment by MiriamNZ** [comment]
Thoughts on 3 & 4
A set amount until the cap is reached, and the same set amount to refill it.
Perhaps there are two things: 
a goal to reach the cap, wth a set amount, then it just stops. Goal done.
a goal to reach a cap, with a set amount, and keep it there, the set amount comping back into play when the balance drops.
